core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -170,6 +182,143 @@ export class ContributionAndReviewService {
+          const suggestionDetails = {
+            suggestion: suggestion,
+            details: targetIdToDetails[suggestion.target_id]
+          };
+          responseSuggestionIdToDetails[suggestion.suggestion_id] = suggestionDetails;
+          });
+          return {
+            suggestionIdToDetails: responseSuggestionIdToDetails,
+            more: false
+          };
+        })
+    );
+  }
+
+  // Function to sort translation cards by state.
+  sortTranslationSuggestionsByState(
+      // eslint-disable-next-line
+      translationSuggestions: any[],
+      states: States,
+      initStateName: string | null
+      // eslint-disable-next-line
+  ): any[] {
+    // Obtain the state names in the order of content flow in the lesson.
+    if (!initStateName) {
+      return translationSuggestions;
+    }
+
+    const stateNamesInOrder = this.computeGraphService.
+      computeBfsTraversalOfStates(
+        initStateName,
+        states,
+        initStateName
+      );
+    //  Create an empty map to store translation cards for each state.
+    const translationSuggestionsByState = (
+      // eslint-disable-next-line
+      new Map<string, any[]>());
+
+    // Assign translation cards to the corresponding state in the map.
+    for (const translationSuggestion of translationSuggestions) {
+      const stateName = translationSuggestion.change.state_name;
+      const suggestionsForState = (
+        translationSuggestionsByState.get(stateName) || []);
+      suggestionsForState.push(translationSuggestion);
+      translationSuggestionsByState.set(stateName, suggestionsForState);
+    }
+
+    // Sort translation cards within each state based on the specified criteria
+    // in the order object.
+    for (const stateName of stateNamesInOrder) {
+      const cardsForState = translationSuggestionsByState.get(stateName) || [];
+
+      cardsForState.sort((cardA, cardB) => {
+        return this.compareTranslationSuggestions(cardA, cardB);
+      });
+      translationSuggestionsByState.set(stateName, cardsForState);
+    }
+
+    // Concatenate the lists of translation cards to create a sorted list.
+    const sortedTranslationCards: SuggestionBackendDict[] = [];
+    for (const stateName of stateNamesInOrder) {
+      const cardsForState = translationSuggestionsByState.get(stateName) || [];
+      sortedTranslationCards.push(...cardsForState);
+    }
+    return sortedTranslationCards;
+  }
+
+  // eslint-disable-next-line
+  private compareTranslationSuggestions(cardA: any, cardB: any): number {
+    const getTypeOrder = (contentId: string): number => {
+      const type = contentId.split('_')[0];
+      const order: { [key: string]: number } = {
+        'content': 0,
+        'interaction': 1,
+        'feedback': 2,
+        'default': 3,
+        'hints': 4,
+        'solution': 5
+      };
+      return order.hasOwnProperty(type) ?
+      order[type] : Number.MAX_SAFE_INTEGER;
+    };
```

Comment:
Can also make this its own function for readability.

Reply:
Done